### ISSUE TITLE
Recma plugins support

### ISSUE THREAD

**Issue by pomber** [facebook/docusaurus#10237]
**TL;DR:** Docusaurus supports remark and rehype plugins, but not recma plugins.

---

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

I'm working on an example showing how to use [Code Hike v1](https://v1.codehike.org/) + Docusaurus, but Docusaurus is missing the option to pass recma plugins that Code Hike needs. 

### Reproducible demo

https://github.com/code-hike/examples/pull/8

### Actual behavior

The `MDXOptions` from `@docusaurus/mdx-loader`:

```ts
export type MDXOptions = {
  admonitions: boolean | Partial<AdmonitionOptions>;
  remarkPlugins: MDXPlugin[];
  rehypePlugins: MDXPlugin[];
  beforeDefaultRemarkPlugins: MDXPlugin[];
  beforeDefaultRehypePlugins: MDXPlugin[];
};
```

### Expected behavior

```ts
export type MDXOptions = {
  admonitions: boolean | Partial<AdmonitionOptions>;
  remarkPlugins: MDXPlugin[];
  rehypePlugins: MDXPlugin[];
  recmaPlugins: MDXPlugin[]; // <-- code hike needs this
  beforeDefaultRemarkPlugins: MDXPlugin[];
  beforeDefaultRehypePlugins: MDXPlugin[];
};
```

`recmaPlugins` is already supported by the `ProcessorOptions` from `@mdx-js/mdx` v3.0.0 that `@docusaurus/mdx-loader` is using.

### Your environment

- Docusaurus version used: 3.4.0

**Comment by slorber** [comment]
👍 that should be easy to add

What I understand is that you need plugins (docs, blog, pages) to support a `recmaPlugins` option as well right? It's not just the mdx loader parameters.

**Comment by pomber** [comment]
> What I understand is that you need plugins (docs, blog, pages) to support a `recmaPlugins` option as well right? It's not just the mdx loader parameters.

Yes! I clicked (go to definition) on the remarkPlugins option from the docusaurus config and it took me directly to the MDXOptions type. That's why I put it here.
